roles/fstat/tasks/main.yml

```diff
@@ -1,17 +1,41 @@
 ---
+- user:
+    name: "{{user}}"
+
 - name: Install required packages
   package:
     state: present
     name: "{{ item }}"
   with_items:
-  - python-virtualenv
-  - gcc
-  - git
+    - python-virtualenv
+    - gcc
+    - git
 
-# TODO create the directory /fstat, do the git checkout
-# automate update
-# set configuration file, and database creation
+# TODO set configuration file, and database creation
 # and selinux boolean
+
+- name: create fstat directory
+  file:
+    path: /fstat/code state=directory
+
+- name: Check if virtualenv exists
+  local_action: stat path=/fstat/env
+  register: env
+
+- name: create virtualenv if it does not exists
+  command: virtualenv /fstat/env
```

Comment:
You should use creates ( https://docs.ansible.com/ansible/latest/command_module.html )

Reply:
So I think that's not exactly clear on what I meant, but that's using creates: /fstat/venv so the task is not run each time.